src/components/sets/SetsManagerActionBar.js

```diff
@@ -0,0 +1,124 @@
+import React from 'react';
+import Ajv from 'ajv';
+import { Icon } from 'antd';
+import 'antd/es/icon/style/index.css';
+import PopoverMenu from './PopoverMenu';
+import SetUnionSVG from '../../assets/sets/union.svg';
+import SetIntersectionSVG from '../../assets/sets/intersection.svg';
+import SetComplementSVG from '../../assets/sets/complement.svg';
+
+import hierarchicalSetsSchema from '../../schemas/hierarchical-sets.schema.json';
+
+export default function SetsManagerActionBar(props) {
+  const { setsTree, datasetId, setsType } = props;
+
+  function onUnion() {
+    const checkedUnion = setsTree.getUnion(setsTree.checkedKeys);
+    setsTree.setCurrentSet(checkedUnion, true, 'Current union');
+    setsTree.emitVisibilityUpdate();
+  }
+
+  function onIntersection() {
+    const checkedIntersection = setsTree.getIntersection(setsTree.checkedKeys);
+    setsTree.setCurrentSet(checkedIntersection, true, 'Current intersection');
+    setsTree.emitVisibilityUpdate();
+  }
+
+  function onComplement() {
+    const checkedComplement = setsTree.getComplement(setsTree.checkedKeys);
+    setsTree.setCurrentSet(checkedComplement, true, 'Current complement');
+    setsTree.emitVisibilityUpdate();
+  }
+
+  function onImport() {
+    const uploadInputNode = document.createElement('input');
+    uploadInputNode.setAttribute('type', 'file');
+    uploadInputNode.setAttribute('accept', 'application/json');
+    document.body.appendChild(uploadInputNode); // required for firefox
+    uploadInputNode.click();
+    uploadInputNode.addEventListener('change', (event) => {
+      if (!(window.File && window.FileReader && window.FileList && window.Blob)) {
+        console.warn('The File APIs are not fully supported in this browser.');
+        return;
+      }
+      const { files } = event.target;
+      if (!files || files.length !== 1) {
+        console.warn('Incorrect number of files selected.');
+        return;
+      }
+      const reader = new FileReader();
+      reader.addEventListener('load', () => {
+        const { result } = reader;
+        const importData = JSON.parse(result);
+        // Validate the imported file.
+        const validate = new Ajv().compile(hierarchicalSetsSchema);
+        const valid = validate(importData);
+        if (!valid) {
+          const failureReason = JSON.stringify(validate.errors, null, 2);
+          console.warn('import validation failed', failureReason);
+        } else if (importData.datasetId !== datasetId) {
+          console.warn('datasetId for import does not match the current datasetId');
```

Comment:
This would be a really easy mistake for an end-user to make, so we need to make sure the error message is also visible to end users.

Reply:
Ok. I have made this commit that passes the error to `STATUS_WARN` https://github.com/hubmapconsortium/vitessce/pull/330/commits/238ea1eb6e114c8233a99e5c11658349a81d4d7a.